**Data preprocessing**

importing libraries for preprocessing 

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("rawdata.csv")
df

id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  gluc  \
0          0  18393       2     168    62.0    110     80            1     1   
1          1  20228       1     156    85.0    140     90            3     1   
2          2  18857       1     165    64.0    130     70            3     1   
3          3  17623       2     169    82.0    150    100            1     1   
4          4  17474       1     156    56.0    100     60            1     1   
...      ...    ...     ...     ...     ...    ...    ...          ...   ...   
69995  99993  19240       2     168    76.0    120     80            1     1   
69996  99995  22601       1     158   126.0    140     90            2     2   
69997  99996  19066       2     183   105.0    180     90            3     1   
69998  99998  22431       1     163    72.0    135     80            1     2   
69999  99999  20540       1     170    72.0    120     80            2     1   

       smoke  alco  active  cardio  
0          0     0       1       0  
1          0     0       1       1  
2          0     0       0       1  
3          0     0       1       1  
4          0     0       0       0  
...      ...   ...     ...     ...  
69995      1     0       1       0  
69996      0     0       1       1  
69997      0     1       0       1  
69998      0     0       0       1  
69999      0     0       1       0  

[70000 rows x 13 columns]

Change age in years

In [3]:
df["age"]=df["age"]/365
df

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
69995  99993  52.712329       2     168    76.0    120     80            1   
69996  99995  61.920548       1     158   126.0    140     90            2   
69997  99996  52.235616       2     183   105.0    180     90            3   
69998  99998  61.454795       1     163    72.0    135     80            1   
69999  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69995     1      1     0       1       0  
69996     2      0     0       1       1  
69997     1      0     1       0       1  
69998     2      0     0       0       1  
69999     1      0     0       1       0  

[70000 rows x 13 columns]

In [4]:
df.to_csv("1_dataset.csv", index=False)

In [5]:
df=pd.read_csv("1_dataset.csv")
df

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
69995  99993  52.712329       2     168    76.0    120     80            1   
69996  99995  61.920548       1     158   126.0    140     90            2   
69997  99996  52.235616       2     183   105.0    180     90            3   
69998  99998  61.454795       1     163    72.0    135     80            1   
69999  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69995     1      1     0       1       0  
69996     2      0     0       1       1  
69997     1      0     1       0       1  
69998     2      0     0       0       1  
69999     1      0     0       1       0  

[70000 rows x 13 columns]

In [6]:
df.sort_values("age")

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
22343  31922  29.583562       2     175    92.0    100     60            1   
30666  43842  29.750685       1     159    59.0    120     80            1   
6219    8850  29.802740       1     175    59.0    120     80            1   
55905  79749  30.038356       1     160    59.0    110     70            1   
31762  45362  39.109589       1     164   105.0    150    110            3   
...      ...        ...     ...     ...     ...    ...    ...          ...   
36603  52297  64.895890       2     163    55.0    120     80            3   
20931  29892  64.904110       1     165    63.0    150     90            2   
50714  72373  64.909589       2     167    68.0    120     80            2   
68005  97124  64.934247       1     178    75.0    120     80            1   
57191  81643  64.967123       2     175    75.0    120     80            1   

       gluc  smoke  alco  active  cardio  
22343     1      0     0       1       0  
30666     1      0     0       1       0  
6219      1      0     0       1       0  
55905     1      0     0       1       0  
31762     3      0     0       1       1  
...     ...    ...   ...     ...     ...  
36603     1      1     0       1       1  
20931     1      0     0       1       1  
50714     2      0     0       0       1  
68005     1      0     0       1       1  
57191     1      0     0       1       1  

[70000 rows x 13 columns]

age values are valid

In [7]:
df.sort_values("height")

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
22723  32456  64.071233       1      55    81.0    130     90            1   
66643  95141  51.589041       1      57    61.0    130     90            1   
64115  91523  50.482192       1      59    57.6    125     67            1   
29157  41661  52.295890       1      60    69.0    110     70            1   
27603  39462  57.473973       1      64    61.0    130     70            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
40090  57307  58.084932       1     198    99.0    140     80            1   
8897   12687  61.635616       2     198    79.0    160    100            1   
41901  59847  52.567123       2     200   100.0    170   1100            2   
21628  30894  52.202740       2     207    78.0    100     70            1   
6486    9223  58.136986       1     250    86.0    140    100            3   

       gluc  smoke  alco  active  cardio  
22723     1      0     0       1       1  
66643     1      0     0       1       1  
64115     1      0     0       0       0  
29157     1      0     0       0       0  
27603     1      0     0       1       0  
...     ...    ...   ...     ...     ...  
40090     1      0     0       1       1  
8897      1      1     0       0       1  
41901     2      0     0       1       1  
21628     1      0     1       1       0  
6486      1      0     0       1       1  

[70000 rows x 13 columns]

55 and 207 are extreme values, lets condense the data set to make it logical

In [8]:
heightfilter= ((df["height"]<=200) & (df["height"]>=100))
df[heightfilter]

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
69995  99993  52.712329       2     168    76.0    120     80            1   
69996  99995  61.920548       1     158   126.0    140     90            2   
69997  99996  52.235616       2     183   105.0    180     90            3   
69998  99998  61.454795       1     163    72.0    135     80            1   
69999  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69995     1      1     0       1       0  
69996     2      0     0       1       1  
69997     1      0     1       0       1  
69998     2      0     0       0       1  
69999     1      0     0       1       0  

[69969 rows x 13 columns]

In [9]:
df[heightfilter].to_csv('1_dataset.csv', index=False)

In [10]:
df = pd.read_csv("1_dataset.csv")
df.sort_values("height")

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
58841  84021  40.219178       2     100    60.0    100     60            1   
17065  24396  58.243836       1     100    86.0    110     70            1   
3419    4838  39.769863       1     100    70.0    100     70            1   
59598  85128  48.336986       1     104   165.0    120     80            1   
45337  64802  54.339726       1     104   159.0    200    100            3   
...      ...        ...     ...     ...     ...    ...    ...          ...   
54263  77425  58.673973       2     198   110.0    140     80            2   
30109  43083  63.764384       1     198    85.0    110     90            1   
57502  82105  51.572603       2     198    78.0    120     80            1   
50189  71657  45.673973       1     198    75.0    170    100            2   
41881  59847  52.567123       2     200   100.0    170   1100            2   

       gluc  smoke  alco  active  cardio  
58841     1      0     0       1       0  
17065     1      0     0       1       1  
3419      1      0     0       1       0  
59598     1      0     0       1       0  
45337     1      0     0       0       1  
...     ...    ...   ...     ...     ...  
54263     3      0     0       1       1  
30109     1      0     0       1       0  
57502     1      1     0       1       1  
50189     1      0     1       1       1  
41881     2      0     0       1       1  

[69969 rows x 13 columns]

In [11]:
weightfilter= ((df["weight"]>=0) & (df["weight"]<=140) )
df[weightfilter]

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
69964  99993  52.712329       2     168    76.0    120     80            1   
69965  99995  61.920548       1     158   126.0    140     90            2   
69966  99996  52.235616       2     183   105.0    180     90            3   
69967  99998  61.454795       1     163    72.0    135     80            1   
69968  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69964     1      1     0       1       0  
69965     2      0     0       1       1  
69966     1      0     1       0       1  
69967     2      0     0       0       1  
69968     1      0     0       1       0  

[69875 rows x 13 columns]

In [12]:
df[weightfilter].to_csv('1_dataset.csv', index=False)

In [13]:
df = pd.read_csv("1_dataset.csv")
df

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
69870  99993  52.712329       2     168    76.0    120     80            1   
69871  99995  61.920548       1     158   126.0    140     90            2   
69872  99996  52.235616       2     183   105.0    180     90            3   
69873  99998  61.454795       1     163    72.0    135     80            1   
69874  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69870     1      1     0       1       0  
69871     2      0     0       1       1  
69872     1      0     1       0       1  
69873     2      0     0       0       1  
69874     1      0     0       1       0  

[69875 rows x 13 columns]

In [14]:
sysfilter= ((df["ap_hi"]>=40) & (df["ap_hi"]<=200) )
df[sysfilter]

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
69870  99993  52.712329       2     168    76.0    120     80            1   
69871  99995  61.920548       1     158   126.0    140     90            2   
69872  99996  52.235616       2     183   105.0    180     90            3   
69873  99998  61.454795       1     163    72.0    135     80            1   
69874  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69870     1      1     0       1       0  
69871     2      0     0       1       1  
69872     1      0     1       0       1  
69873     2      0     0       0       1  
69874     1      0     0       1       0  

[69590 rows x 13 columns]

In [15]:
df[sysfilter].to_csv('1_dataset.csv', index=False)

In [16]:
df = pd.read_csv("1_dataset.csv")
df

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
69585  99993  52.712329       2     168    76.0    120     80            1   
69586  99995  61.920548       1     158   126.0    140     90            2   
69587  99996  52.235616       2     183   105.0    180     90            3   
69588  99998  61.454795       1     163    72.0    135     80            1   
69589  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69585     1      1     0       1       0  
69586     2      0     0       1       1  
69587     1      0     1       0       1  
69588     2      0     0       0       1  
69589     1      0     0       1       0  

[69590 rows x 13 columns]

In [17]:
dysfilter= ((df["ap_lo"]>=40) & (df["ap_lo"]<=150))
df[dysfilter]

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
69585  99993  52.712329       2     168    76.0    120     80            1   
69586  99995  61.920548       1     158   126.0    140     90            2   
69587  99996  52.235616       2     183   105.0    180     90            3   
69588  99998  61.454795       1     163    72.0    135     80            1   
69589  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69585     1      1     0       1       0  
69586     2      0     0       1       1  
69587     1      0     1       0       1  
69588     2      0     0       0       1  
69589     1      0     0       1       0  

[68588 rows x 13 columns]

In [18]:
df[dysfilter].to_csv('1_dataset.csv', index=False)

In [19]:
df = pd.read_csv("1_dataset.csv")
df

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
68583  99993  52.712329       2     168    76.0    120     80            1   
68584  99995  61.920548       1     158   126.0    140     90            2   
68585  99996  52.235616       2     183   105.0    180     90            3   
68586  99998  61.454795       1     163    72.0    135     80            1   
68587  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
68583     1      1     0       1       0  
68584     2      0     0       1       1  
68585     1      0     1       0       1  
68586     2      0     0       0       1  
68587     1      0     0       1       0  

[68588 rows x 13 columns]

The filters applied help keep the data logically possible

Creating a new column called diagnosis, default value is 0

In [20]:
df["diagnosis"]=0
df

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
68583  99993  52.712329       2     168    76.0    120     80            1   
68584  99995  61.920548       1     158   126.0    140     90            2   
68585  99996  52.235616       2     183   105.0    180     90            3   
68586  99998  61.454795       1     163    72.0    135     80            1   
68587  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  diagnosis  
0         1      0     0       1       0          0  
1         1      0     0       1       1          0  
2         1      0     0       0       1          0  
3         1      0     0       1       1          0  
4         1      0     0       0       0          0  
...     ...    ...   ...     ...     ...        ...  
68583     1      1     0       1       0          0  
68584     2      0     0       1       1          0  
68585     1      0     1       0       1          0  
68586     2      0     0       0       1          0  
68587     1      0     0       1       0          0  

[68588 rows x 14 columns]

In [21]:
df.to_csv('2_dataset.csv', index=False)

In [22]:
df =pd.read_csv("2_dataset.csv")
df

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
68583  99993  52.712329       2     168    76.0    120     80            1   
68584  99995  61.920548       1     158   126.0    140     90            2   
68585  99996  52.235616       2     183   105.0    180     90            3   
68586  99998  61.454795       1     163    72.0    135     80            1   
68587  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  diagnosis  
0         1      0     0       1       0          0  
1         1      0     0       1       1          0  
2         1      0     0       0       1          0  
3         1      0     0       1       1          0  
4         1      0     0       0       0          0  
...     ...    ...   ...     ...     ...        ...  
68583     1      1     0       1       0          0  
68584     2      0     0       1       1          0  
68585     1      0     1       0       1          0  
68586     2      0     0       0       1          0  
68587     1      0     0       1       0          0  

[68588 rows x 14 columns]

In [23]:
x= df["ap_hi"]
y= df["ap_lo"]


# syshypo=int(115)
# sysnormal=int(125)

# diahypo=int(75)
# dianormal=int(85)



for i in df.index:
    if ((int (x[i])<int(115)) & (int (y[i])< int(80))):
        df["diagnosis"][i]= 1
        
    elif ((int (x[i])> int(130)) & (int (y[i])> int(80))) :
        df["diagnosis"][i]= 3
    else:
        df["diagnosis"][i]= 2  
        
df.value_counts("diagnosis")      
        
df.to_csv('2_dataset.csv', index=False)

C:\Users\Darshan\AppData\Local\Temp\ipykernel_22936\4129809799.py:20: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["diagnosis"][i]= 2
C:\Users\Darshan\AppData\Local\Temp\ipykernel_22936\4129809799.py:18: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["diagnosis"][i]= 3
C:\Users\Darshan\AppData\Local\Temp\ipykernel_22936\4129809799.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["diagnosis"][i]= 1


In [24]:
df=pd.read_csv("2_dataset.csv")
df

id        age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0  50.391781       2     168    62.0    110     80            1   
1          1  55.419178       1     156    85.0    140     90            3   
2          2  51.663014       1     165    64.0    130     70            3   
3          3  48.282192       2     169    82.0    150    100            1   
4          4  47.873973       1     156    56.0    100     60            1   
...      ...        ...     ...     ...     ...    ...    ...          ...   
68583  99993  52.712329       2     168    76.0    120     80            1   
68584  99995  61.920548       1     158   126.0    140     90            2   
68585  99996  52.235616       2     183   105.0    180     90            3   
68586  99998  61.454795       1     163    72.0    135     80            1   
68587  99999  56.273973       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  diagnosis  
0         1      0     0       1       0          2  
1         1      0     0       1       1          3  
2         1      0     0       0       1          2  
3         1      0     0       1       1          3  
4         1      0     0       0       0          1  
...     ...    ...   ...     ...     ...        ...  
68583     1      1     0       1       0          2  
68584     2      0     0       1       1          3  
68585     1      0     1       0       1          3  
68586     2      0     0       0       1          2  
68587     1      0     0       1       0          2  

[68588 rows x 14 columns]

In [25]:
df=df.astype(int)
df.dtypes

id             int32
age            int32
gender         int32
height         int32
weight         int32
ap_hi          int32
ap_lo          int32
cholesterol    int32
gluc           int32
smoke          int32
alco           int32
active         int32
cardio         int32
diagnosis      int32
dtype: object

In [26]:
df.to_csv("3_dataset.csv",  index= False)

In [27]:
df = pd.read_csv("3_dataset.csv")
df

id  age  gender  height  weight  ap_hi  ap_lo  cholesterol  gluc  \
0          0   50       2     168      62    110     80            1     1   
1          1   55       1     156      85    140     90            3     1   
2          2   51       1     165      64    130     70            3     1   
3          3   48       2     169      82    150    100            1     1   
4          4   47       1     156      56    100     60            1     1   
...      ...  ...     ...     ...     ...    ...    ...          ...   ...   
68583  99993   52       2     168      76    120     80            1     1   
68584  99995   61       1     158     126    140     90            2     2   
68585  99996   52       2     183     105    180     90            3     1   
68586  99998   61       1     163      72    135     80            1     2   
68587  99999   56       1     170      72    120     80            2     1   

       smoke  alco  active  cardio  diagnosis  
0          0     0       1       0          2  
1          0     0       1       1          3  
2          0     0       0       1          2  
3          0     0       1       1          3  
4          0     0       0       0          1  
...      ...   ...     ...     ...        ...  
68583      1     0       1       0          2  
68584      0     0       1       1          3  
68585      0     1       0       1          3  
68586      0     0       0       1          2  
68587      0     0       1       0          2  

[68588 rows x 14 columns]

In [28]:
df.value_counts("diagnosis")

diagnosis
2    44686
3    14519
1     9383
dtype: int64